x-pack/filebeat/module/cisco/amp/ingest/pipeline.yml

```diff
@@ -300,21 +309,41 @@ processors:
     value: "{{ cisco.amp.computer.external_ip }}"
     if: ctx?.cisco?.amp?.computer?.external_ip != null
     allow_duplicates: false
-- foreach:
-    field: cisco.amp.computer.network_addresses
-    processor:
-      append:
-        field: related.ip
-        value: "{{ _ingest._value.ip }}"
-        allow_duplicates: false
+- script:
+    lang: painless
+    source: |
+        if (ctx?.related == null) {
+            ctx.related = new HashMap();
+            ctx.related.ip = new ArrayList();
+        }
+        if (ctx?.related != null && ctx?.related?.ip == null) {
+            ctx.related.ip = new ArrayList();
+        }
+        for (addr in ctx?.cisco?.amp?.computer?.network_addresses) {
+            if (addr.ip != null && addr.ip != "") {
+                if (!ctx?.related?.ip.contains(addr.ip)) {
+                    ctx?.related?.ip.add(addr.ip);  
+                }
+            }
+        }
     if: ctx?.cisco?.amp?.computer?.network_addresses != null
-- foreach:
-    field: cisco.amp.computer.network_addresses
-    processor:
-      append:
-        field: cisco.amp.related.mac
-        value: "{{ _ingest._value.mac }}"
-        allow_duplicates: false
+- script:
+    lang: painless
+    source: |
+        if (ctx?.cisco?.amp?.related == null) {
+            ctx.cisco.amp.related = new HashMap();
+            ctx.cisco.amp.related.mac = new ArrayList();
+        }
+        if (ctx?.cisco?.amp?.related != null && ctx?.cisco?.amp?.related?.mac == null) {
+            ctx.cisco.amp.related.mac = new ArrayList();
+        }
+        for (addr in ctx?.cisco?.amp?.computer?.network_addresses) {
+            if (addr.mac != null && addr.mac != "") {
+                if (!ctx?.cisco?.amp?.related?.mac.contains(addr.mac)) {
+                    ctx?.cisco?.amp?.related?.mac.add(addr.mac);  
+                }
+            }
+        }
     if: ctx?.cisco?.amp?.computer?.network_addresses != null
 - foreach:
     field: cisco.amp.vulnerabilities
```

Comment:
on a separate note, I would change line 352 from `cisco.amp.related.cve` to the ECS field of `vulnerability.id`

Reply:
I would really like to do the same, unfortunately the field is currently not a array as per the ECS definition, it might move or be added as a alias to the ECS field if the specification will allow it in the future.